pkg/reportingplugins/mercury/types.go

```diff
@@ -41,3 +41,16 @@ type Transmitter interface {
 	// - FromAccount() should return CSA public key
 	ocrtypes.ContractTransmitter
 }
+
+type ChainReader interface {
+	// Head returns the current chain head
+	Head(context.Context) (Block, error)
+	// LatestBlocks returns an ordered list of the latest specified number of blocks
+	LatestBlocks(context.Context, int) ([]Block, error)
+}
```

Comment:
I'm still looking this over, but wanted to draw attention to this PR where we define `type ChainReader interface` as a part of PluginProvider:
https://github.com/smartcontractkit/chainlink-relay/pull/196

We should coordinate between these to make sure we don't have too many merge conflicts or conflicting definitions.

Reply:
What did you have in mind? Including this on the shared `ChainReader` was ruled out of scope, so we are staying out of the way by working in the encapsulated mercury provider.